dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+    }
+
+    UpdateItemsSource();
+}
+
+void Breadcrumb::OnPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    const winrt::IDependencyProperty& property = args.Property();
+
+    // TODO: Implement
+    if (property == s_ItemsSourceProperty)
+    {
+        UpdateItemsSource();
+    }
+    else if (property == s_ItemTemplateProperty)
+    {
+        UpdateItemTemplate();
+    }
+    else if (property == s_DropdownItemTemplateProperty)
+    {
+        UpdateDropdownItemTemplate();
+    }
+}
+
+void Breadcrumb::OnBreadcrumbItemRepeaterLoaded(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    if (const auto& breadcrumbItemRepeater = m_breadcrumbItemRepeater.get())
+    {
+        OnRepeaterCollectionChanged(nullptr, nullptr);
+    }
+}
+
+void Breadcrumb::UpdateItemTemplate()
+{
+    const winrt::IInspectable& newItemTemplate = ItemTemplate();
+    m_breadcrumbElementFactory->UserElementFactory(newItemTemplate);
+}
+
+void Breadcrumb::UpdateDropdownItemTemplate()
+{
+    const winrt::IInspectable& newItemTemplate = DropdownItemTemplate();
+
+    // Copy the item template to the ellipsis button too
+    if (const auto& ellipsisBreadcrumbItem = m_ellipsisBreadcrumbItem.get())
+    {
+        if (const auto& itemImpl = winrt::get_self<BreadcrumbItem>(ellipsisBreadcrumbItem))
+        {
+            itemImpl->SetFlyoutDataTemplate(newItemTemplate);
+        }
+    }
+}
+
+void Breadcrumb::UpdateItemsSource()
+{
+    m_itemsSourceChanged.revoke();
+    m_itemsSourceChanged2.revoke();
+
+    m_breadcrumbItemsRepeaterItemsSource = winrt::ItemsSourceView(ItemsSource());
+    m_itemsSourceChanged = m_breadcrumbItemsRepeaterItemsSource.CollectionChanged(winrt::auto_revoke, { this, &Breadcrumb::OnRepeaterCollectionChanged });
+
+    if (const auto& breadcrumbItemRepeater = m_breadcrumbItemRepeater.get())
+    {
+        if (const auto& itemsSource = this->ItemsSource())
+        {
+            const auto& incc = [this]() {
+                if (this->ItemsSource())
+                {
+                    return this->ItemsSource().try_as<winrt::INotifyCollectionChanged>();
+                }
+                else
+                {
+                    return this->ItemsSource().try_as<winrt::INotifyCollectionChanged>();
+                }
+            }();
+
+            if (incc)
+            {
+                m_collectionChanged = incc.CollectionChanged({ this, &Breadcrumb::OnRepeaterCollectionChanged });
+                m_notifyCollectionChanged.set(incc);
+            }
+
+            if (const auto& collection = itemsSource.try_as<winrt::IObservableVector<IInspectable>>())
+            {
+                m_itemsSourceChanged2 = collection.VectorChanged(winrt::auto_revoke, { this, &Breadcrumb::OnRepeaterCollectionChanged });
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnRepeaterCollectionChanged(const winrt::IInspectable&, const winrt::IInspectable& args)
```

Comment:
For consistency, should use ItemsRepeater everywhere, and not Repeater or ItemRepeater.